cpp/CMakeLists.txt

```diff
@@ -980,6 +980,7 @@ endif()
 
 if(ARROW_JNI)
   add_subdirectory(src/jni)
+  add_subdirectory(../java/dataset/src/main/cpp ./java/jni)
```

Comment:
Similar to the other PR, this still needs to be conditioned on `if(ARROW_DATASET)` right?

Reply:
done